pageserver/compaction/src/helpers.rs

```diff
@@ -129,12 +153,12 @@ impl<'a, E: CompactionJobExecutor> PartialOrd for LazyLoadLayer<'a, E> {
 impl<'a, E: CompactionJobExecutor> Ord for LazyLoadLayer<'a, E> {
     fn cmp(&self, other: &Self) -> std::cmp::Ordering {
         // reverse order so that we get a min-heap
-        other.key().cmp(&self.key())
+        (other.key(), other.lsn()).cmp(&(self.key(), self.lsn()))
```

Comment:
I'm start wondering why Heikki didn't include LSN before. Is the `E::Key` expected to be only the key, or should it be Key + LSN instead?

Reply:
also, this implementation is giving a reverse order

Reply:
> I'm start wondering why Heikki didn't include LSN before. Is the E::Key expected to be only the key, or should it be Key + LSN instead?

From how I understand tiered compaction, it will always decrease the key range for delta layers, so tiered compaction will not produce stacked delta layers.

It's possible that this has led to the assumption that there is no lsn-stacked delta layers serving as inputs to compaction, which is wrong: when in memory layers get turned into deltas, and there is no compaction for a long enough time for this to happen multiple times, we obtain stacked delta layers (0000-FFF in key range and increasing lsn ranges) which then get fed to compaction.